### PR TITLE
Metaprogram methods into anonymous module to fix respond_to

### PR DESCRIPTION
**Pull request by rreinhardt9**
## Why

When I created a policy that subclassed from the base policy and wanted to write tests to see if my new policy responded to what I thought it should, `respond_to?` returned false even though the policy actually did respond to the action in question. I created a `grade` action_on_object and asked if it responded to `#grade?` and it returned false even though it actually worked.

## What

First possibility I saw was to make sure to define a `responds_to_missing?` to go along with our `method_missing` as seen [here](https://makandracards.com/makandra/9821-when-overriding-method_missing-remember-to-override-respond_to_missing-as-well).

But I was curious if we could just eliminate method_missing. The main catch there is that we wanted a way that we could override the actions in a base class without using the `alias_method` do-si-do and if we had a method of the same name in the base class, then `define_method` would not define the method in the super class because it was already defined... and therefore we would get an error when trying to call `super` in the override method.

To solve that, it sounds like we can use modules to place the methods higher in the inheritance hierarchy than the subclass... allowing us to define the methods straight in the module, override them in a subclass while calling super, and they would also return true when calling `respond_to?` because we would be able to ditch method_missing. Move info on this technique in [this blog post](https://thepugautomatic.com/2013/07/dsom/)

## Caveats

I removed defining `list` when an action is a general action... because we don't list general actions.

If we like this method, may want to document those methods a little better to make it clear what is going on.

### CODE REVIEW

**Comment by rreinhardt9** [comment]
@armilam @ehourigan I'd love your thought on if this would be a good direction for us. If so, I can polish up the pr for real and figure out the test suite issue.

**Comment by armilam** [comment]
Hey, if this works, I love a good simplification!

**Comment by rreinhardt9** [comment]
see [ch16727](https://app.clubhouse.io/lessonly/story/16727/accessly-does-not-respond-to-respond-to)

File: lib/accessly/policy/base.rb [hunk 1]
```diff
@@ -1,24 +1,77 @@
...
+      #
+      # @param actions [Hash] the actions to define on the policy
+      #
+      # @example
+      #
+      #   actions(
+      #     some_action: 1,
+      #     flip_the_flop: 2,
+      #     create: 3
+      #   )
```
[the remark this answers is not in the document]
Comment (rreinhardt9) [reply lib/accessly/policy/base.rb]: I love it, great idea!!! I'll add that documentation for this and the other method 💯

**Comment by rreinhardt9** [comment]
screen shot 2018-10-11 at 3 44 12 pm

I updated the documentation @armilam , thanks for the idea! Here is a view of what it looks like rendered through yard. I've compressed those updated down into the original commit.

**Comment by armilam** [comment]
Well, CI is technically failing, but that's only because I was trying to get the tests running and they aren't yet. This is good to merge.

**Pull request merged**